cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1031,29 +1031,27 @@ def ceph_storage_privilege():
     reconfigure_apiserver(endpoint_from_flag('etcd.available'))
 
 
+@when('ceph-client.connected')
 @when('kubernetes-master.ceph.configured')
 @when_not('kubernetes-master.ceph.pool.created')
-def ceph_storage_pool():
+def ceph_storage_pool(ceph_client):
     '''Once Ceph relation is ready,
     we need to add storage pools.
 
     :return: None
     '''
     hookenv.log('Creating Ceph pools.')
 
-    ReplicatedPool(
-        name='xfs-pool',
-        service='admin',
-        replicas=3,
-        app_name=None,
-    ).create()
+    pools = [
+        'xfs-pool',
+        'ext4-pool'
+    ]
 
-    ReplicatedPool(
-        name='ext4-pool',
-        service='admin',
-        replicas=3,
-        app_name=None,
-    ).create()
+    for pool in pools:
+        ceph_client.create_pool(
```

Comment:
Do we want to set a status message and wait on the flag to be set telling us the pools are created? What if the pool creation fails? Do we get notification of that in some way so we can surface an error to the user?

Reply:
Good point!  I'll add logging.